### ISSUE TITLE
Reflection Probe differences between Always and Once cause Materials to look radically different

### ISSUE THREAD

**Issue by Menithal** [godotengine/godot#82177]
### Godot version

v4.1.1.stable.mono.official [bd6af8e0e]

### System information

Godot v4.1.1.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated AMD Radeon RX 7900 XTX (Advanced Micro Devices, Inc.; 31.0.21031.1005) - AMD Ryzen 7 5800X3D 8-Core Processor (16 Threads)

### Issue description

![AlwaysVsOnce](https://github.com/godotengine/godot/assets/11814599/d185ec90-569f-48d5-96b1-f06b26d99a3e)

When I add an reflection probe to my node tree, my materials start looking off. If I play around with the settings of the reflection probe's and set update mode from Once to Always, the materials look correct and one to one with how they'd appear in Blender Eevee. (see above vs  below)
![BlenderTest](https://github.com/godotengine/godot/assets/11814599/aa1648e2-ac11-43d2-a0df-ab3797bb1b32)

If set to the Reflection Probe to Once,
 the roughness maps feel like are sampled up to intervals of 0.125-0.25. (from 0.0-1.0), and skipping gradients which causes stuff that shouldn't be so reflective to be very reflective. This happens both with the Standard and ORM version of the godot default materials.

Expected:

Once vs Always should behave similarly, even if one of them is updated when called/ modified vs the other every time the frame is updated.

### Steps to reproduce

1. Create scene with an matrice of balls with incrementing values of roughness and metallicness, a simple hdri environment image, and a reflection probe.
2. Observe Result.
3. Swap Reflection Probe update mode to Always
4. Observe Result.
5. Compare Results, see difference.

### Minimal reproduction project

[ReflectionProbeTest.zip](https://github.com/godotengine/godot/files/12705931/ReflectionProbeTest.zip)

**Comment by bitsawer** [comment]
Seems like disabling ReflectionProbe cull mask layer 1 makes the behavior more similar between the two update modes. Alternatively, setting Ambient mode to Disabled has some effect, too. I'm not an expert on ReflectionProbe, so I can't say if the behavior is intentional or a bug.

**Comment by Menithal** [comment]
> Seems like disabling ReflectionProbe cull mask layer 1 makes the behavior more similar between the two update modes. Alternatively, setting Ambient mode to Disabled has some effect, too. I'm not an expert on ReflectionProbe, so I can't say if the behavior is intentional or a bug.

Yeah, this would be making it to equal to as if the Reflection probe wasnt there.
The intention though is to have the reflection probe however, so it still is a bug  unfortunately.


More precise would be "Once" mode for Reflection Probe does not match the material characteristics if the Reflection Probe wasnt there, or set to Always.

**Comment by Calinou** [comment]
This is expected, as the way the roughness mipmaps are updated differs when using the Always update mode for performance reasons.

When using the Once update mode, the high-quality filter is used to generate roughness mipmaps across 8 frames or so.
When using the Always update mode, the real-time filter is used to generate roughness mipmaps in a single frame.
